### PR TITLE
Copy labels as specified by copy_labels_pattern

### LINKED ISSUE #87: Port origin PR labels to backport PR
Allow a set of labels to be automatically applied onto the backport PR.

Use case for us (NixOS/nixpkgs) would be labels like `Severity: Security`, which also applies to the backport PR and right now needs to be remembered to be applied when the PR gets created.

### PR DESCRIPTION
**Pull request by lheckemann**
Fixes #87

### CODE REVIEW

**Comment by lheckemann** [comment]
@korthout I tried rebasing on main to fix the merge conflict, but it appears that `main` doesn't currently build:

```
$ npm run build

> backport-action@1.1.0-SNAPSHOT build
> tsc

src/git.ts:1:10 - error TS2616: 'execa' can only be imported by using 'import execa = require("execa")' or a default import.

1 import { execa } from "execa";
           ~~~~~


Found 1 error in src/git.ts:1
```

File: README.md [hunk 1]
```diff
@@ -67,6 +67,11 @@ jobs:
           # These are indicated by a dollar sign and curly braces (`${placeholder}`).
           # Please refer to this action's README for all available placeholders.
           # pull_title: "[Backport ${target_branch}] ${pull_title}"
+
+          # Optional
+          # Regex pattern to match github labels which will be copied from the
+          # original PR to the backport PR.
```
Comment (korthout) [inline README.md:73]: Let's clarify what happens to the backport labels. Also please note that the documentation generally uses the term `pull request` over `PR`.
```suggestion
          # Regex pattern to match github labels which will be copied from the
          # original pull request to the backport pull request. Labels matching
          # `label_pattern` will not be copied.
```

Please also apply these changes to `action.yml`

File: README.md [hunk 2]
```diff
@@ -67,6 +67,11 @@ jobs:
...
+          # copy_labels_pattern: ^severity.*
```
Comment (korthout) [inline README.md:74]: The examples show the default value. I'd like to continue that.
```suggestion
          # copy_labels_pattern: null
```

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks so much @lheckemann 🚀 Such a cool feature. I'm sure my team wants to use this as well.

🙇 I have 2 small change requests, and then we should be able to merge this. Let me know if you want to apply these, or whether I should take it over. Happy to do so.

>I tried rebasing on main to fix the merge conflict, but it appears that `main` doesn't currently build

The `exaca` dependency was updated to a new major version with a breaking change (the import is now named, instead of default). Running `npm ci` locally should solve the issue for you. You can then build a new version with `npm run all`.

File: README.md (hunk 2 above)
Comment (lheckemann) [reply README.md:74]: Can we still put a "useful" example in the docs?

Comment (korthout) [reply README.md:74]: Sure. How about in the comment which describes the field?

**Comment by korthout** [comment]
@lheckemann Would you like me to take it over from here, or do you want to work on the requested changes yourself?

**Comment by lheckemann** [comment]
I noticed another problem though: the default of `null` probably doesn't trigger the no-regex-defined logic and will instead probably result in all labels being copied. I dug around in this a couple days ago and didn't really get that far, partly for lack of familiarity with GHA, so if you're willing to take that over I'd appreciate it! Otherwise I'm sure I'll get around to bashing my head against this again within a couple of weeks.

**Comment by korthout** [comment]
My apologies, @lheckemann. I've prioritized #317 because it was already fully functional (and smaller in scope). I'll be out of the office next week, but I'll continue with this pull request as soon as I find the time again.

**Comment by korthout** [comment]
@lheckemann I've just pushed a new branch [`korthout:copy-labels`](https://github.com/korthout/backport-action/tree/copy-labels) rebasing your PR on top of `main` and potentially fixing the problem you described above.

Next, I want to run some tests with this to see if all works well. But it looks good so far! 🎉

**Comment by lheckemann** [comment]
Oh, that's a simple solution! Thanks for taking care of it.

**Comment by korthout** [comment]
I had to shift focus a bit before I could release it, but it's released in `v1.2.0` 🚀 

Thanks again for all your efforts @lheckemann 🥇

**Pull request closed without merging**